src/main/java/uk/nhs/digital/nhsconnect/nhais/repository/SchedulerTimestamp.java

```diff
@@ -0,0 +1,22 @@
+package uk.nhs.digital.nhsconnect.nhais.repository;
+
+import java.time.Instant;
+
+import lombok.Data;
+
+import org.springframework.data.mongodb.core.index.CompoundIndex;
+import org.springframework.data.mongodb.core.index.CompoundIndexes;
+import org.springframework.data.mongodb.core.mapping.Document;
+
+@CompoundIndexes({
+    @CompoundIndex(
+        name = "unique_document",
+        def = "{'updateTimestamp' : 1, 'schedulerType': 1}",
```

Comment:
Should the index only include 'schedulerType' to ensure there is only a single document per scheduler type?

Reply:
I think so, good point